R/tidy_ard_order.R

```diff
@@ -0,0 +1,79 @@
+#' Standard Column Order of ARD
+#'
+#' Relocate all columns in ARD to the standard order.
+#'
+#' @param x (`data.frame`)\cr
+#'   a ARD data frame of class 'card'
+#'
+#' @return a data frame
+#' @export
+#'
+#' @examples
+#' ard <-
+#'   dplyr::bind_rows(
+#'     ard_continuous(mtcars, variables = "mpg"),
+#'     ard_continuous(mtcars, variables = "mpg", by = "cyl")
+#'   )
+#' names(ard)
+#'
+#' tidy_ard_column_order(ard)
+tidy_ard_column_order <- function(x) {
+  group_cols <- dplyr::select(x, all_ard_groups()) |> names() |> sort()
+
+  dplyr::select(
+    x,
+    all_of(group_cols),
+    all_ard_variables(),
+    any_of(c(
+      "context",
+      "stat_name", "stat_label", "statistic", "statistic_fmt", "statistic_fmt_fn",
+      "warning", "error"
+    )),
+    dplyr::everything()
+  )
+}
+
+
+#' Order rows of ARD
+#'
+#' Order rows of ARD according to variables, groups, and strata, retaining the
+#' order of input ARDs
+#'
+#' @param x (`data.frame`)\cr
+#'   a ARD data frame of class 'card'
+#'
+#' @return a data frame
+#' @export
+#'
+#' @examples
+#' ard <-
+#'   bind_ard(
+#'     ard_continuous(mtcars, variables = c("mpg","disp"), by = "cyl"),
+#'     ard_ttest(mtcars, variable = "mpg", by = "cyl"),
+#'     ard_ttest(mtcars, variable = "disp", by = "cyl")
+#'   )
+#' tidy_ard_row_order(ard)
```

Comment:
Since the default of `bind_ard()` is to order, this isn't doing any ordering.

Reply:
Ahhh thank you for catching this!